rafthttp/stream.go

```diff
@@ -407,3 +447,38 @@ func isClosedConnectionError(err error) bool {
 	operr, ok := err.(*net.OpError)
 	return ok && operr.Err.Error() == "use of closed network connection"
 }
+
+// serverVersion returns the version from the given header.
+func serverVersion(h http.Header) *semver.Version {
+	verStr := h.Get("X-Server-Version")
+	// backward compatibility with etcd 2.0
+	if verStr == "" {
+		verStr = "2.0.0"
+	}
+	return semver.Must(semver.NewVersion(verStr))
+}
+
+// mainVersion returns a version that has the same major and minor value
+// as the given one, and sets patch and additional label to empty.
+func mainVersion(v *semver.Version) *semver.Version {
```

Comment:
It only cares about major and minor version when we compare two versions and check the version support because major and minor decides the functionality, and patch cannot add new functionality.